omnigibson/utils/python_utils.py

```diff
@@ -590,20 +591,20 @@ def __getitem__(self, item):
     def __setitem__(self, key, value):
         self.add_fcn(name=key, fcn=value)
 
-    def get(self, name, *args, **kwargs):
+    def get(self, name):
         """
         Computes the function referenced by @name with the corresponding @args and @kwargs. Note that for a unique
         set of arguments, this value will be internally cached
 
         Args:
             name (str): The name of the function to call
-            *args (tuple): Positional arguments to pass into the function call
-            **kwargs (tuple): Keyword arguments to pass into the function call
 
         Returns:
             any: Output of the function referenced by @name
         """
-        return self._fcns[name](*args, **kwargs)
+        if name not in self._cache:
+            self._cache[name] = self._fcns[name]()
+        return self._cache[name]
```

Comment:
I removed this on purpose: the caching currently already implicitly happens in ControllableObjectViewAPI thanks to its batching mechanism so this actually doesn't do anything except make cache misses just a bit slower.

Reply:
Oh interesting...that makes sense because I was almost positive I wouldn't have missed such a glaring bug lol

When I was timing I found that something is evidently not caching properly in ControllableObjectViewAPI, as the time cost is linear when called multiple times during the same controller loop, which doesn't seem correct.